Formatting CLI Tool (#15664)
ballerina-platform/ballerina-lang: misc/ballerina-formatter/README.md

# Ballerina Format CLI Tool

> Ballerina format CLI tool is the tool that can be used to format the ballerina sources

Reviewer: ```suggestion
The Ballerina format CLI tool can be used to format Ballerina sources
```

> according to the default [Ballerina style guide](https://github.com/ballerina-platform/ballerina-lang/tree/master/docs/style_guide)

Reviewer: ```suggestion
according to the default [Ballerina style guide](https://github.com/ballerina-platform/ballerina-lang/tree/master/docs/style_guide).
```

> ## Using Ballerina Format CLI tool

Reviewer: ```suggestion
## Using the Ballerina Format CLI Tool
```

> Ballerina format command can be used to format ballerina source files.

Reviewer: ```suggestion
The Ballerina format command can be used to format Ballerina source files.
```

```sh
$ ballerina format --help

> Get format tool help page.

Reviewer: ```suggestion
Displays the help page of the format tool.
```

Usage:
ballerina format [<ballerinaFile> | <moduleName>] [-d | --dry-run]
  ballerinaFile:

>       Paths to a single ballerina source file which needs to be formatted.

Reviewer: ```suggestion
      Directs to a single Ballerina source file, which needs to be formatted.
```

  moduleName:

>       Name of the module which has ballerina source files which need to

Reviewer: ```suggestion
      The name of the module, which has Ballerina source files that need to
```

>       be formatted. Command should be executed from the ballerina project root.

Reviewer: ```suggestion
      be formatted. The format command should be executed from the root of the Ballerina project.
```

Flags:
  -d
  --dry-run 

>       by providing this option user can dry run formatter and see which files will

Reviewer: ```suggestion
      By providing this option, you can dry run the formatter and see which files will
```

>       be formatted after execution.

Reviewer: ```suggestion
      be formatted after the execution.
```

```

### Usage Examples

> **Example 1:** Format all the Ballerina source files in a Ballerina project.

Reviewer: ```suggestion
**Example 1:** Formats all the Ballerina source files in a Ballerina project.
```

```sh
$ ballerina format
```

> This command should be executed from the Ballerina project root.

Reviewer: ```suggestion
This command should be executed from the root of the Ballerina project.
```

> **Example 2:** Format all the Ballerina source files in a Ballerina module.

Reviewer: ```suggestion
**Example 2:** Formats all the Ballerina source files in a Ballerina module.
```

```sh
$ ballerina format module1
```

> This command should be executed from the Ballerina project root.

Reviewer: ```suggestion
This command should be executed from the root of the Ballerina project.
```

> **Example 3:** Format a single Ballerina source file.

Reviewer: ```suggestion
**Example 3:** Formats a single Ballerina source file.
```

```sh
$ ballerina format hello.bal
```

> **Example 4:** Dry run the formatter to see which files will be formatted

Reviewer: ```suggestion
**Example 4:** Performs a dry run of the formatter to see which files will be formatted
```
